components/proxy/src/main/java/com/hotels/styx/startup/extensions/PluginLoadingForStartup.java

```diff
@@ -115,10 +116,19 @@ public MetricRegistry metricRegistry() {
                 return environment.metricRegistry().scope(DEFAULT_PLUGINS_METRICS_SCOPE + "." + factory.name());
             }
 
+            /**
+             * @deprecated deprecated in favor of {@link PluginFactory.Environment#pluginMeterRegistry()}
+             */
+            @Deprecated
```

Comment:
I understand why you have done this. But the fact that you need to indicates that the inheritance hierarchy is not right. If a plugin should not have access to `meterRegistry()` (except through the new class, as I've suggested above) then the plugin environment should not inherit this method.